.eslintrc.js

```diff
@@ -18,15 +17,39 @@ module.exports = {
       },
     ],
     'require-jsdoc': 'error',
-
-    'self/meta-property-ordering': 'off',
-    'self/require-meta-docs-url': 'off',
-    'self/report-message-format': ['error', '^[^a-z].*.$'],
   },
   overrides: [
+    {
+      // Apply eslint-plugin rules to our own rules/tests (but not docs).
+      files: ['lib/**/*.js', 'tests/**/*.js'],
+      plugins: ['self'],
+      extends: ['plugin:self/all'],
+      rules: {
+        'self/meta-property-ordering': 'off',
+        'self/report-message-format': ['error', '^[^a-z].*.$'],
+        'self/require-meta-docs-url': 'off',
+      },
+    },
     {
       files: ['tests/**/*.js'],
       env: { mocha: true },
     },
+    {
+      files: ['**/*.md'],
+      processor: 'markdown/markdown',
+    },
+    {
+      // Markdown JS code samples in documentation:
+      files: ['**/*.md/*.js'],
+      plugins: ['markdown'],
+      parserOptions: {
+        sourceType: 'module',
+      },
```

Comment:
`sourceType` should be "scripts".

Reply:
Removed this config so it would default to `script`.